### ISSUE TITLE
EPUBCheck Error after inserting Author

### ISSUE THREAD

**Issue by shorshe** [Sigil-Ebook/Sigil#176]
I'm stumbling over an epubcheck error with sigil 0.9.2 from now and then. I'm editing epubs, which are exported from Indesign.

After inserting an author via the Sigil metadata editor I get the following epubcheck (4.0.1) error:

```
ERROR	RSC-005	/OEBPS/content.opf
(line 9, col 28)	Error while parsing file 'attribute "role" not allowed here; expected attribute "id", "ns:file-as", "ns:role" or "xml:lang" (with xmlns:ns="http://www.idpf.org/2007/opf")'.

ERROR	RSC-005	/OEBPS/content.opf
(line 13, col 35)	Error while parsing file 'attribute "event" not allowed here; expected attribute "id", "ns1:event" or "ns2:type" (with xmlns:ns1="http://www.idpf.org/2007/opf" xmlns:ns2="http://www.w3.org/2001/XMLSchema-instance")'.
```
This is the metadata section before the change (valid EPUB)
```
  <metadata>
    <dc:identifier id="bookid">978-3-86248-344-0</dc:identifier>
    <meta content="Adobe InDesign" name="generator"/>
    <meta name="cover" content="tanalyse_khan.jpg"/>
    <meta content="0.9.2" name="Sigil version"/>
    <dc:title>Technische-Analyse – Klar und einfach</dc:title>
    <dc:language>de</dc:language>
    <dc:rights>© 2013 by FinanzBuch Verlag, ein Imprint der Münchner Verlagsgruppe GmbH, Nymphenburger Straße 86, D-80636 München</dc:rights>
    <dc:publisher>FinanzBuch Verlag</dc:publisher>
    <dc:date opf:event="modification" xmlns:opf="http://www.idpf.org/2007/opf">2016-01-20</dc:date>
  </metadata>
```

This is the metadata section after the change (invalid EPUB)
```
  <metadata>
    <dc:identifier id="bookid">978-3-86248-344-0</dc:identifier>
    <meta content="Adobe InDesign" name="generator"/>
    <meta name="cover" content="tanalyse_khan.jpg"/>
    <meta content="0.9.2" name="Sigil version"/>
    <dc:title>Technische-Analyse – Klar und einfach</dc:title>
    <dc:creator role="aut">Michael Khan</dc:creator>
    <dc:language>de</dc:language>
    <dc:rights>© 2013 by FinanzBuch Verlag, ein Imprint der Münchner Verlagsgruppe GmbH, Nymphenburger Straße 86, D-80636 München</dc:rights>
    <dc:publisher>FinanzBuch Verlag</dc:publisher>
    <dc:date event="modification">2016-01-20</dc:date>
    <dc:date opf:event="modification" xmlns:opf="http://www.idpf.org/2007/opf">2016-01-20</dc:date>
  </metadata>
```

**Comment by kevinhendricks** [comment]
If you are going to use extensions  for dc:creator metadata (the role attribute is not an official part of the dc:creator spec) need to set set the attribute namespace for the extensions you want in the metadata tag itself first.

So your metadata tag should look like the following, before editing the metadata in the gui.

'''
<metadata xmlns:opf="http://www.idpf.org/2007/opf">
'''

Then when you edit with the gui, it will create the following:

'''
<dc:creator opf:role="aut" opf:scheme="marc-relators">Michael Khan</dc:creator>
'''

The way you had it set, the opf namespace was set in the package tag to have no prefix, which is fine everyplace but in the metadata tag itself.  I will look to see if Sigil is mot properly adding it for some reason.

The easiest solution is to simply hand edit the content.opf metadata as above.  It should pass all checks then.

**Comment by kevinhendricks** [comment]
With the correct metadata xmlns attribute set, I can not recreate your issue.  So I am closing this issue.

**Comment by shorshe** [comment]
Just as a quick note. This also happens when I just insert new images into the ebook without changing anything other than that. 
After saving epubcheck shows errors and I have to correct the metadata tag by hand a then reenter the other meta information.

**Comment by kevinhendricks** [comment]
Is this an epub2 or an epub3?  If an epub2, then I simply can not recreate this.  If an epub3, the yes Sigil-0.9.2 and Sigil-0.9.3 both have issues when adding existing files to an epub3, it is messing up the metadata.  After adding files, you will need to hand correct the metadata. This is all fixed in Sigil master (the forthcoming Sigil-0.9.4) and a new epub2/epub3 metadata gui editor has been added which prevents the conflict in metadata between epub2 and epub3. It also has a large number of other bugs fixed which may be impacting the situation.  We are waiting on updated translations and then should release it quite soon (barring any new bug reports).

So please give that a try when it is released and let me know if it doesn't fix your issues by posting here.

Thanks,

KevinH

**Comment by shorshe** [comment]
Hi Kevin,

It's a (valid) EPUB2. I can send you a sample if you like. 
This Issue seems to be solved with 0.94.

**Comment by kevinhendricks** [comment]
Glad to hear it is fixed in Sigil-0.9.4.

Thanks for your bug report.
